packages/wasm-parser/test/index.js

```diff
@@ -5,52 +5,77 @@ const diff = require("jest-diff");
 const { NO_DIFF_MESSAGE } = require("jest-diff/build/constants");
 const { writeFileSync, readFileSync } = require("fs");
 const path = require("path");
-
+const wabt = require("wabt");
+const { parse } = require("@webassemblyjs/wast-parser");
 const { decode } = require("../lib");
+const { traverse } = require("@webassemblyjs/ast");
 
 function toArrayBuffer(buf) {
   return buf.buffer.slice(buf.byteOffset, buf.byteOffset + buf.byteLength);
 }
 
-function createCheck(suite, ast) {
-  const expectedFile = path.join(path.dirname(suite), "expected.json");
-  const code = JSON.stringify(ast, null, 2);
+function jsonTraverse(o, func) {
+  func(o);
+  for (const i in o) {
+    if (o[i] !== null && typeof o[i] == "object") {
+      jsonTraverse(o[i], func);
+    }
+  }
+}
+
+// remove the additional metadata from the wasm parser
+function stripMetadata(ast) {
+  jsonTraverse(ast, node => {
+    delete node.metadata;
+    delete node.loc;
+  });
+  traverse(ast, {
+    // currently the WAT parser does not support function type definitions
+    TypeInstruction(path) {
+      path.remove();
+    }
+  });
+  return ast;
+}
 
-  let expected;
-  try {
-    expected = readFileSync(expectedFile, "utf8");
-    expected = JSON.stringify(JSON.parse(expected), null, 2);
-  } catch (e) {
-    expected = code;
+function createCheck(actualWatPath) {
+  // convert the wat fixture to wasm
+  const actualWat = readFileSync(actualWatPath, "utf8");
+  const module = wabt.parseWat(actualWatPath, actualWat);
+  const { buffer } = module.toBinary({ write_debug_names: true });
+  const actualWasmPath = path.join(path.dirname(actualWatPath), "actual.wasm");
+  writeFileSync(actualWasmPath, new Buffer(buffer));
 
-    writeFileSync(expectedFile, code);
+  // read the wasm file and strip custom metadata
+  const bin = toArrayBuffer(readFileSync(actualWasmPath));
+  const ast = stripMetadata(decode(bin));
+  const actual = JSON.stringify(ast, null, 2);
 
-    console.log("Write expected file", expectedFile);
-  }
+  // parse the wat file to create the expected AST
+  const astFromWat = parse(actualWat);
+  const expected = JSON.stringify(astFromWat, null, 2);
 
-  const out = diff(code.trim(), expected.trim());
+  const out = diff(expected.trim(), actual.trim());
 
   if (out !== null && out !== NO_DIFF_MESSAGE) {
     throw new Error("\n" + out);
   }
 
   // When one line the error is not caught
-  if (code.trim() !== expected.trim()) {
+  if (actual.trim() !== expected.trim()) {
     throw new Error("Assertion error");
   }
 }
 
 describe("compiler", () => {
   describe("Binary format parsing", () => {
     const testSuites = glob.sync(
-      "packages/wasm-parser/test/fixtures/**/actual.wasm"
+      "packages/wasm-parser/test/fixtures/block/**/actual.wat"
```

Comment:
Just a reminder to remove this later and technically they are `wast` files.

Reply:
good spot - thanks